### PR TITLE
[SPARK-16886][EXAMPLES][DOC] Fix some examples to be consistent and indentation in documentation

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

Originally this PR was based on #14491 but I realised that fixing examples are more sensible rather than comments. 

This PR fixes three things below:
- Fix two wrong examples in `structured-streaming-programming-guide.md`. Loading via `read.load(..)` without `as` will be `Dataset<Row>` not `Dataset<String>` in Java.
- Fix indentation across `structured-streaming-programming-guide.md`. Python has 4 spaces and Scala and Java have double spaces. These are inconsistent across the examples. 
- Fix `StructuredNetworkWordCountWindowed` and  `StructuredNetworkWordCount` in Java and Scala to initially load `DataFrame` and `Dataset<Row>` to be consistent with the comments and some examples in `structured-streaming-programming-guide.md` and to match Scala and Java to Python one (Python one loads it as `DataFrame` initially).
## How was this patch tested?

N/A

Closes https://github.com/apache/spark/pull/14491

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
Hi @srowen , Could you please take a look?

File: examples/src/main/java/org/apache/spark/examples/sql/streaming/JavaStructuredNetworkWordCountWindowed.java [hunk 1]
```diff
@@ -75,28 +75,30 @@ public static void main(String[] args) throws Exception {
...
       .readStream()
       .format("socket")
       .option("host", host)
       .option("port", port)
       .option("includeTimestamp", true)
-      .load().as(Encoders.tuple(Encoders.STRING(), Encoders.TIMESTAMP()));
+      .load();
 
     // Split the lines into words, retaining timestamps
-    Dataset<Row> words = lines.flatMap(
```
Comment (HyukjinKwon) [inline examples/src/main/java/org/apache/spark/examples/sql/streaming/JavaStructuredNetworkWordCountWindowed.java]: This diff looks a lot but the actual change is only

```java
 -      .load().as(Encoders.tuple(Encoders.STRING(), Encoders.TIMESTAMP()));
 +      .load();
```

and

```java
- Dataset<Row> words = lines.flatMap(
+ Dataset<Row> words = lines
+  .as(Encoders.tuple(Encoders.STRING(), Encoders.TIMESTAMP()))
+  .flatMap(
```

**Comment by srowen** [comment]
OK, seems reasonable.

**Comment by srowen** [comment]
LGTM

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**